### PR TITLE
Add STPAdditionalSourceInfo

### PR DESCRIPTION
**Pull request by bg-stripe**
r? @bdorfman-stripe 

- moves `metadata` to from UserInformation to AdditionalSourceInfo
- adds `pollingTimeout` to PaymentConfiguration
- updates the iDEAL `requestPayment` test to verify UserInfo / AdditionalSourceInfo get used correctly

### CODE REVIEW

File: Stripe/PublicHeaders/STPAdditionalSourceInfo.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ *  You can retrieve this metadata from the webhook event, and use it to fulfill
+ *  your customer's order.
+ *  https://stripe.com/docs/sources#best-practices
+ */
+@property(nonatomic, copy, nullable)NSDictionary<NSString *, NSString *>*metadata;
+
+/**
+ *  Some payment methods, like SOFORT, allow setting a custom statement descriptor.
+ *  By default, your Stripe account’s statement descriptor is used (you can review 
+ *  this in the Dashboard at https://dashboard.stripe.com/account).
```
Comment (bdorfman-stripe) [inline Stripe/PublicHeaders/STPAdditionalSourceInfo.h]: Maybe make more clear that this only works for specific source and can't generally alter your statement descriptor, with a link to the sources docs,

File: Stripe/PublicHeaders/STPAdditionalSourceInfo.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ *  Metadata associated with your user. This will be attached to any Source
+ *  objects created by `STPPaymentContext`, `STPAddSourceViewController`, etc.
+ *  You should consider storing any order information (e.g., order number) here.
+ *  For payment methods that require additional user action, your backend will
+ *  need to listen to the `source.chargeable` webhook to create a charge request.
+ *  You can retrieve this metadata from the webhook event, and use it to fulfill
+ *  your customer's order.
+ *  https://stripe.com/docs/sources#best-practices
+ */
+@property(nonatomic, copy, nullable)NSDictionary<NSString *, NSString *>*metadata;
```
Comment (bdorfman-stripe) [inline Stripe/PublicHeaders/STPAdditionalSourceInfo.h]: Should probably add that you should store customer ID here for webhook based sources so you can pass that through when the charge is made? Or maybe that belongs somewhere else (in our future "how to webhook source in iOS" doc page).

File: Stripe/PublicHeaders/STPPaymentConfiguration.h [hunk 3]
```diff
@@ -116,6 +116,15 @@ NS_ASSUME_NONNULL_BEGIN
...
 @property (nonatomic, nullable, copy) NSURL *returnURL  NS_EXTENSION_UNAVAILABLE("Redirect based sources are not available in extensions");
 
+
+/**
+ *  For payment methods that redirect a user to authorize a payment, the SDK
+ *  will poll the source for up to this amount of time after the user returns to
+ *  the app in order to determine the status of the payment. The default is 10
+ *  seconds.
+ */
+@property (nonatomic) NSTimeInterval pollingTimeout;
```
Comment (bdorfman-stripe) [inline Stripe/PublicHeaders/STPPaymentConfiguration.h]: There's a built in maximum to the poller timeout right? Should mention that here as well, or refer to that other api doc.

Comment (bg-stripe) [inline Stripe/PublicHeaders/STPPaymentConfiguration.h]: ah yep, will add a note

File: Stripe/PublicHeaders/STPAdditionalSourceInfo.h (hunk 1 above)
Comment (bg-stripe) [inline Stripe/PublicHeaders/STPAdditionalSourceInfo.h]: ah yep, I'll update

File: Stripe/PublicHeaders/STPAdditionalSourceInfo.h (hunk 2 above)
Comment (bg-stripe) [inline Stripe/PublicHeaders/STPAdditionalSourceInfo.h]: Let's wait on the followup to our ask – sounds like we might end up associating the customer for them?

Comment (bdorfman-stripe) [inline Stripe/PublicHeaders/STPAdditionalSourceInfo.h]: Sounds good

**Comment by bg-stripe** [comment]
@bdorfman-stripe updated in 890204a

**Comment by bdorfman-stripe** [comment]
👍

**Pull request merged**